### PR TITLE
removed equal sign from env buffer, added defensive guard against the key

### LINKED ISSUE #61: k8s hook: equal signs (=) in environment vars  are escaped in job pods
while using k8s container mode with ARC, I found 'google-github-actions/auth@v1'  action stopped working and the problem was '=' in base64 strings were escaped ('\\=') and that broke the JSON parsing.

the following workflow can show this problem.

```yaml
name: K8S runner env issue
env:
  foo: '{"user": "YWJjCg=="}'
on:
  workflow_dispatch:
jobs:
  envissue:
    runs-on: k8s
    container:
      image: ubuntu
    steps:
    - run: |
        apt-get update
        apt-get install -y jq
        echo "foo: $foo"        
        echo "$foo" | jq -r .user

```
This workflow will fail on k8s runners (with k8s hook)
Actual output is :

```
foo: {"user": "YWJjCg\=\="}
parse error: Invalid escape at line 1, column 21
```

expected output is:

```
foo: {"user": "YWJjCg=="}
YWJjCg==
```

### PR DESCRIPTION
**Pull request by nikola-jokic**
Fixes #61

### CODE REVIEW

File: packages/k8s/src/k8s/utils.ts [hunk 1]
```diff
@@ -111,11 +111,11 @@ export function writeEntryPointScript(
   if (environmentVariables && Object.entries(environmentVariables).length) {
     const envBuffer: string[] = []
     for (const [key, value] of Object.entries(environmentVariables)) {
+      if (key.includes('=') || key.includes(`'`) || key.includes('"')) {
```
Comment (johnsudol) [inline packages/k8s/src/k8s/utils.ts:114]: Small Nit: Might be better to make all of them template strings for readability sake
```suggestion
      if (key.includes(`=`) || key.includes(`'`) || key.includes(`"`)) {
```

**Review by johnsudol — APPROVED** [review]
Left one small nit, but otherwise LGTM!

**Review by johnsudol — DISMISSED** [review]

File: packages/k8s/src/k8s/utils.ts (hunk 1 above)
Comment (nikola-jokic) [reply packages/k8s/src/k8s/utils.ts]: I agree, thanks :smile:

File: packages/k8s/src/k8s/utils.ts [hunk 2]
```diff
@@ -111,11 +111,11 @@ export function writeEntryPointScript(
   if (environmentVariables && Object.entries(environmentVariables).length) {
     const envBuffer: string[] = []
     for (const [key, value] of Object.entries(environmentVariables)) {
+      if (key.includes(`=`) || key.includes(`'`) || key.includes(`"`)) {
+        throw new Error(`environment key ${key} is invalid`)
```
Comment (fhammerl) [inline packages/k8s/src/k8s/utils.ts:115]:
```suggestion
        throw new Error(`environment key ${key} is invalid - the key must not contain =, ' or "`)
```

NIT: If this exception can be hit on genuine user error, we could be more explicit on our rules - if you don't want to expose the implementation it's fine too.

**Review by fhammerl — DISMISSED** [review]
LGTM

**Review by johnsudol — APPROVED** [review]

**Pull request merged**